### PR TITLE
Add missing return types in 4.0

### PR DESCRIPTION
**Pull request by smnandre**
Revert the removal of return types done in #3156 

Should i propagate to `extra` packages in the same PR ?

### CODE REVIEW

File: src/Extension/ExtensionInterface.php [hunk 1]
```diff
@@ -31,35 +31,35 @@ interface ExtensionInterface
      *
      * @return TokenParserInterface[]
      */
-    public function getTokenParsers();
+    public function getTokenParsers(): array;
```
Comment (Kocal) [inline src/Extension/ExtensionInterface.php:34]: Should we also update the PHPDoc return type to `array<TokenParserInterface>`, or keeping `TokenParserInterface[]` is fine (`array<>` is more strict)?

Comment (stof) [reply src/Extension/ExtensionInterface.php:34]: it is not more strict. It is strictly equivalent.

Comment (Kocal) [reply src/Extension/ExtensionInterface.php:34]: Ah, I misread/misremembered https://phpstan.org/writing-php-code/phpdoc-types#general-arrays

**Comment by fabpot** [comment]
@smnandre Yes, we should do it on `extra` as well.

**Comment by smnandre** [comment]
I made a pass on them and think it's good :)

**Comment by fabpot** [comment]
Thank you @smnandre.

**Comment by smnandre** [comment]
Arf I found others while checking sorry ....  in code runtimeloader + docs. I'll open another PR :|

**Pull request merged**